### PR TITLE
Add getting help section with links to SO and Gitter

### LINKED ISSUE #707: Create a "sponsored" open chat group
## Overview

I have felt that since joining the Kotlin Slack community I have been able to absorb the language from maintainers and the community much faster than by trying to locate these resources myself.
I believe that having an active group like this will benefit extension authors, engine developers, test writers, and other consumers of JUnit.

I would like to see the the JUnit 5 team adopt a Slack/Gitter for both the community and maintainers to gather around.

## Deliverables

- [x] A ~~Slack~~/**Gitter**/~~other~~ chat platform
- [ ] README documentation updated

### PR DESCRIPTION
**Pull request by sormuras**
## Overview

Add getting help section to README.md file and the User Guide with links to SO and Gitter.

Closes #707

---

I hereby agree to the terms of the JUnit Contributor License Agreement.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,6 +31,9 @@ to file feature requests and bug reports using the project's
 <a href="https://github.com/junit-team/junit5/issues?q=is%3Aissue+is%3Aopen+label%3Aup-for-grabs">`up-for-grabs`</a>
 label are specifically targeted for community contributions.
 
+## Getting Help
+Ask JUnit 5 related questions at [StackOverflow] or chat with us at [Gitter].
```
Comment (marcphilipp) [inline README.md]: "on" StackOverflow/Gitter?

Comment (sbrannen) [inline README.md]: "on" in both cases.

File: documentation/src/docs/asciidoc/overview.adoc [hunk 2]
```diff
@@ -32,8 +32,15 @@ the platform.
...
-JUnit 5 requires Java 8 at runtime. However, you can still test code that has been
-compiled with previous versions of the JDK.
+JUnit 5 requires Java 8 (or higher) at runtime. However, you can still test code that
+has been compiled with previous versions of the JDK.
+
+
+[[overview-getting-help]]
+=== Getting Help
+
+Ask JUnit 5 related questions at {stackoverflow}[StackOverflow] or chat with us at
```
Comment (sbrannen) [inline documentation/src/docs/asciidoc/overview.adoc]: again, "on" in both cases. 😉

**Comment by sormuras** [comment]
4x "on" applied.

File: documentation/src/docs/asciidoc/index.adoc [hunk 3]
```diff
@@ -15,6 +15,8 @@ ifdef::backend-pdf[:imagesdir: {outdir}/{imagesdir}]
 :javadoc-root:                      link:../api
 //
 :snapshot-repo:                     https://oss.sonatype.org/content/repositories/snapshots
+:stackoverflow:                     https://stackoverflow.com/questions/tagged/junit5
+:gitter-chat:                       https://gitter.im/junit-team/junit5
```
Comment (sbrannen) [inline documentation/src/docs/asciidoc/index.adoc]: For consistency, these new links should be placed (alphabetically) at the bottom alongside the links to AssertJ, etc.

File: documentation/src/docs/asciidoc/overview.adoc [hunk 4]
```diff
@@ -32,8 +32,15 @@ the platform.
...
-compiled with previous versions of the JDK.
+JUnit 5 requires Java 8 (or higher) at runtime. However, you can still test code that
+has been compiled with previous versions of the JDK.
+
+
+[[overview-getting-help]]
+=== Getting Help
+
+Ask JUnit 5 related questions on {stackoverflow}[StackOverflow] or chat with us on
+{gitter-chat}[Gitter].
```
Comment (sbrannen) [inline documentation/src/docs/asciidoc/overview.adoc]: For consistency, `[StackOverflow]` and `[Gitter]` should be defined with the corresponding links in `index.adoc`.

File: README.md [hunk 5]
```diff
@@ -128,3 +131,5 @@ See also <http://repo1.maven.org/maven2/org/junit/> for releases and <https://os
 [JUnit 5 User Guide]: http://junit.org/junit5/docs/current/user-guide/
 [Prototype]: https://github.com/junit-team/junit5/wiki/Prototype
 [Twitter]: https://twitter.com/junitlambda
+[StackOverflow]: https://stackoverflow.com/questions/tagged/junit5
+[Gitter]: https://gitter.im/junit-team/junit5
```
Comment (sbrannen) [inline README.md]: For consistency, these new links should be inserted alphabetically into this list.

**Pull request merged**